### PR TITLE
core: post load infra endpoint

### LINKED ISSUE #4005: core: create post loading infra endpoint (trigger)
# Description

Create get /infra/state endpoint in core to get the current state of the loading of the infra. 

# Goal

We want to return the current state of an infra

# Acceptance Criteria

- [ ] the frontend uses the core endpoint
- [ ] The endpoint return the state of the infra

# Description

Create post /infra/{id}/load endpoint in editoast to load the infra. 

# Goal

We want to let the user load the infra. 

# Acceptance Criteria

- [ ] openapi is updated
- [ ] the frontend uses the endpoint
- [ ] The endpoint loads the infrastructure

### PR DESCRIPTION
**Pull request by younesschrifi**
Close #4005

### CODE REVIEW

File: core/openapi.yaml [hunk 1]
```diff
@@ -44,6 +44,27 @@ paths:
                     nullable: true
                 required:
                   - git_describe
+  /infra_load:
+    post:
+      tags:
+        - infra
+      summary: load the infrastructure
+      requestBody:
+        description: Infrastructure id and waypoints
```
Comment (eckter) [inline core/openapi.yaml]: The description doesn't match the content

File: core/openapi.yaml [hunk 2]
```diff
@@ -44,6 +44,27 @@ paths:
                     nullable: true
                 required:
                   - git_describe
+  /infra_load:
+    post:
+      tags:
+        - infra
+      summary: load the infrastructure
```
Comment (Castavo) [inline core/openapi.yaml]: You could be more precise in this summary, for example mention that it loads the infra from editoast

File: core/openapi.yaml (hunk 1 above)
Comment (Castavo) [inline core/openapi.yaml]: I think you forgot to change the description

File: core/src/main/java/fr/sncf/osrd/api/InfraLoadEndpoint.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
+package fr.sncf.osrd.api;
+
+import com.squareup.moshi.Json;
+import com.squareup.moshi.JsonAdapter;
+import com.squareup.moshi.Moshi;
+import fr.sncf.osrd.railjson.parser.exceptions.InvalidRollingStock;
+import fr.sncf.osrd.railjson.parser.exceptions.InvalidSchedule;
+import fr.sncf.osrd.railjson.schema.rollingstock.RJSRollingStock;
+import fr.sncf.osrd.railjson.schema.schedule.RJSStandaloneTrainSchedule;
+import fr.sncf.osrd.railjson.schema.schedule.RJSTrainPath;
```
Comment (eckter) [inline core/src/main/java/fr/sncf/osrd/api/InfraLoadEndpoint.java]:
```suggestion
```

unused

File: core/src/main/java/fr/sncf/osrd/api/InfraLoadEndpoint.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        public String expectedVersion;
+
+
+        /**
+         * Create a default InfraLoadRequest
+         */
+        public InfraLoadRequest() {
+            infra = null;
+            expectedVersion = null;
+        }
```
Comment (eckter) [inline core/src/main/java/fr/sncf/osrd/api/InfraLoadEndpoint.java]:
```suggestion
```

unused

File: core/src/test/java/fr/sncf/osrd/api/InfraLoadingTest.java
Comment (Castavo) [inline core/src/test/java/fr/sncf/osrd/api/InfraLoadingTest.java:1]: You could (maybe shouldn't ?) add more tests to challenge edge cases

**Review by Castavo — COMMENTED** [review]
Nice PR :grin:

**Review by eckter — COMMENTED** [review]
A few minor comments, otherwise LGTM

File: core/src/test/java/fr/sncf/osrd/api/InfraLoadingTest.java
Comment (Castavo) [reply core/src/test/java/fr/sncf/osrd/api/InfraLoadingTest.java:1]: In fact edge cases tests would be for the InfraManager.load method, so they're out of scope in this PR I'd say

**Review by eckter — APPROVED** [review]
